spatula/base.py

```diff
@@ -96,3 +96,43 @@ def convert_pdf(self, filename, type):
         data = pipe.read()
         pipe.close()
         return data
+
+
+class CSV(AbstractPage):
+    def __init__(self, scraper, url=None, *, obj=None, **kwargs):
+        super().__init__(scraper, url=url, obj=obj, **kwargs)
+        resp = self.do_request()
+        self.reader = csv.DictReader(io.StringIO(resp.text))
+
+    def do_request(self):
+        return self.scraper.get(self.url)
+
+    def scrape_page_items(self, page_type, url=None, **kwargs):
```

Comment:
No, `AbstractPage` only has a constructor. I copied these function prototypes from `Page`, which also inherits from `AbstractPage`.

However, looking at this further, I don't think we really need these two methods here. IIUC, `scrape_page_items()` and `scrape_page()` are only used when following a link from one scraped page to another, respectively for pages that will yield multiple objects or return just one. I doubt that many CSVs will merit following links, so I'll take these two methods out now.